src/node.rs

```diff
@@ -1105,8 +1127,21 @@ impl<S: ForkSource + std::fmt::Debug> InMemoryNode<S> {
     }
 
     /// Runs L2 transaction and commits it to a new block.
-    fn run_l2_tx(&self, l2_tx: L2Tx, execution_mode: TxExecutionMode) -> Result<(), String> {
+    fn run_l2_tx(&self, mut l2_tx: L2Tx, execution_mode: TxExecutionMode) -> Result<(), String> {
         let tx_hash = l2_tx.hash();
+        {
+            let inner = self
+                .inner
+                .read()
+                .map_err(|e| format!("Failed to acquire read lock: {}", e))?;
+            if let Some(impersonated_account) = inner.impersonated_account {
+                tracing::info!(
+                    "🕵️ Executing tx from impersonated account {:?}",
+                    impersonated_account
+                );
+                l2_tx = adjust_tx_initiator(l2_tx, impersonated_account);
```

Comment:
I think we can simply override the `l2_tx.common_data.initiator_address` here since `l2_tx` is already defined `mut` above